src/PaymentGateways/PayPalCommerce/PayPalCommerce.php

```diff
@@ -83,5 +84,11 @@ static function() {
 		if ( wp_doing_cron() ) {
 			give( RefreshToken::class );
 		}
+
+		give()->singleton( AdvancedCardFields::class );
+		Hooks::addAction( 'give_paypal-commerce_cc_form', AdvancedCardFields::class, 'addCreditCardForm', 10, 3 );
+
+		give()->singleton( DonationProcessor::class );
+		Hooks::addAction( 'give_gateway_paypal-commerce', DonationProcessor::class, 'handle', 10, 3 );
```

Comment:
Singletons should be added in the Service Provider `register` method. The reason being that by the time things are done (i.e. booting the app) all of the singletons and such should be registered so _any_ service can access them.

Reply:
Also, happy to see the `Hooks:addAction` in the wild! 🙌

Reply:
fixed: https://github.com/impress-org/givewp/compare/d21f485e38a4...fb45ca1e07e3